### ISSUE TITLE
Cannot clip a Multipolygon

### ISSUE THREAD

**Issue by Stepp1** [corteva/rioxarray#94]
#### Code Sample, a copy-pastable example if possible

I used the attached NetCDF file.

```python
>>> xds = xds.rio.set_crs("epsg:4326")
>>> xds.rio.clip(roi_gdf['geometry'][0], "epsg:4326")

---------------------------------------------------------------------------
MissingCRS                                Traceback (most recent call last)
<ipython-input-57-d9c37b3214e5> in <module>
----> 1 xds.rio.clip(roi_gdf['geometry'][0], "epsg:4326")

/opt/tljh/user/lib/python3.7/site-packages/rioxarray/rioxarray.py in clip(self, geometries, crs, all_touched, drop, invert)
   1469         for var in self.vars:
   1470             clipped_dataset[var] = self._obj[var].rio.clip(
-> 1471                 geometries, crs=crs, all_touched=all_touched, drop=drop, invert=invert
   1472             )
   1473         return clipped_dataset

/opt/tljh/user/lib/python3.7/site-packages/rioxarray/rioxarray.py in clip(self, geometries, crs, all_touched, drop, invert)
   1039         if self.crs is None:
   1040             raise MissingCRS(
-> 1041                 "CRS not found. Please set the CRS with 'set_crs()' or 'write_crs()'."
   1042             )
   1043         dst_crs = CRS.from_user_input(crs_to_wkt(crs))

MissingCRS: CRS not found. Please set the CRS with 'set_crs()' or 'write_crs()'.

>> xds = xds.rio.set_spatial_dims('lat', 'lon', inplace=False)

>> xds = xds.rio.write_crs("epsg:4326")

>>> xds.rio.clip(roi_gdf['geometry'][0], "epsg:4326")
---------------------------------------------------------------------------
DimensionError                            Traceback (most recent call last)
<ipython-input-61-d9c37b3214e5> in <module>
----> 1 xds.rio.clip(roi_gdf['geometry'][0], "epsg:4326")

/opt/tljh/user/lib/python3.7/site-packages/rioxarray/rioxarray.py in clip(self, geometries, crs, all_touched, drop, invert)
   1469         for var in self.vars:
   1470             clipped_dataset[var] = self._obj[var].rio.clip(
-> 1471                 geometries, crs=crs, all_touched=all_touched, drop=drop, invert=invert
   1472             )
   1473         return clipped_dataset

/opt/tljh/user/lib/python3.7/site-packages/rioxarray/rioxarray.py in clip(self, geometries, crs, all_touched, drop, invert)
   1050         clip_mask_arr = geometry_mask(
   1051             geometries=geometries,
-> 1052             out_shape=(int(self.height), int(self.width)),
   1053             transform=self.transform(),
   1054             invert=not invert,

/opt/tljh/user/lib/python3.7/site-packages/rioxarray/rioxarray.py in height(self)
    515         if self._height is not None:
    516             return self._height
--> 517         self._height = self._obj[self.y_dim].size
    518         return self._height
    519 

/opt/tljh/user/lib/python3.7/site-packages/rioxarray/rioxarray.py in y_dim(self)
    499             return self._y_dim
    500         raise DimensionError(
--> 501             "x dimension not found. 'set_spatial_dims()' can address this."
    502         )
    503 

DimensionError: x dimension not found. 'set_spatial_dims()' can address this.
```
#### Problem description

I'm trying to clip a MultiPolygon of a netCDF file but rioxarray cannot find the necessary crs information.

Until this point I had been only using `xds.sel`.

I've attached the netCDF file, and the multipolygon's geojson.

[test.nc.zip](https://github.com/corteva/rioxarray/files/4295490/test.nc.zip)
[region_test.geojson.zip](https://github.com/corteva/rioxarray/files/4295491/region_test.geojson.zip)

Ubuntu 18.04.3 and MacOS 10.15.3
Almost all packages are freshly installed through Conda

#### Expected Output

A clipped version of the netCDF file

#### Output of ``rioxarray --version``

<details>
'0.0.22'

</details>

**Comment by snowman2** [comment]
Seems like `set_spatial_dims` is not working as expected. I would recommend doing this:

```python
xds = xarray.open_dataset("test.nc").rename({"lon": "longitude", "lat": "latitude"})
xds = xds.rio.write_crs("EPSG:4326")
xds_clip = xds.rio.clip(gpd['geometry'][0], "EPSG:4326")
```

**Comment by Stepp1** [comment]
Hi, 
Thanks @snowman2! Renaming the coordinates lets me clip data correctly.

I think that it should confirm that `set_spatial_dims` is not working.

**Comment by snowman2** [comment]
Yep - I have a fix in #95. Thanks for the report!

**Comment by snowman2** [comment]
Just released 0.0.23 - will be in conda-forge soon: https://github.com/conda-forge/rioxarray-feedstock/pull/20
